x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/DataFrameFeatureSet.java

```diff
@@ -69,30 +101,118 @@ public boolean enabled() {
+            ClientHelper.DATA_FRAME_ORIGIN,
+            totalTransformCount,
+            totalTransformCountListener,
+            client::search);
     }
 
-    static DataFrameFeatureSetUsage createUsage(boolean available,
-                                                boolean enabled,
-                                                List<DataFrameTransformStateAndStats> transformsStateAndStats) {
+    static DataFrameIndexerTransformStats parseSearchAggs(SearchResponse searchResponse) {
+        List<Long> statisticsList = new ArrayList<>(PROVIDED_STATS.length);
 
-        Map<String, Long> transformsCountByState = new HashMap<>();
-        DataFrameIndexerTransformStats accumulatedStats = new DataFrameIndexerTransformStats();
-        transformsStateAndStats.forEach(singleResult -> {
-            transformsCountByState.merge(singleResult.getTransformState().getIndexerState().value(), 1L, Long::sum);
-            accumulatedStats.merge(singleResult.getTransformStats());
-        });
+        for(String statName : PROVIDED_STATS) {
+            Aggregation agg = searchResponse.getAggregations().get(statName);
+            if (agg instanceof NumericMetricsAggregation.SingleValue) {
+                statisticsList.add((long)((NumericMetricsAggregation.SingleValue)agg).value());
+            } else {
+                statisticsList.add(0L);
+            }
+        }
+        return new DataFrameIndexerTransformStats(null,
+            statisticsList.get(0),  // numPages
+            statisticsList.get(1),  // numInputDocuments
+            statisticsList.get(2),  // numOutputDocuments
+            statisticsList.get(3),  // numInvocations
+            statisticsList.get(4),  // indexTime
+            statisticsList.get(5),  // searchTime
+            statisticsList.get(6),  // indexTotal
+            statisticsList.get(7),  // searchTotal
+            statisticsList.get(8),  // indexFailures
+            statisticsList.get(9)); // searchFailures
+    }
+
+    static void getStatisticSummations(Client client, ActionListener<DataFrameIndexerTransformStats> statsListener) {
+        QueryBuilder queryBuilder = QueryBuilders.constantScoreQuery(QueryBuilders.boolQuery()
+            .filter(QueryBuilders.termQuery(DataFrameField.INDEX_DOC_TYPE.getPreferredName(),
+                DataFrameIndexerTransformStats.NAME)));
+
+        SearchRequestBuilder requestBuilder = client.prepareSearch(DataFrameInternalIndex.INDEX_NAME)
+            .setSize(0)
+            .setQuery(queryBuilder);
+
+        for(String statName : PROVIDED_STATS) {
+            requestBuilder.addAggregation(AggregationBuilders.sum(statName).field(statName));
+        }
 
-        return new DataFrameFeatureSetUsage(available, enabled, transformsCountByState, accumulatedStats);
+        ActionListener<SearchResponse> getStatisticSummationsListener = ActionListener.wrap(
+            searchResponse -> statsListener.onResponse(parseSearchAggs(searchResponse)),
```

Comment:
@hendrikmuhs do you think shardFailures should cause this endpoint to fail? I am not sure partial results should cause a failure here (especially since this endpoint is used in many places). 

What do you think of just logging their failure?

Reply:
I agree, failing would be inconsistent behavior. For this case I think logging makes sense.

For get stats: Maybe we can add it to `List<TaskOperationFailure>`? I think it should be part of the response if possible.